astropy/modeling/mappings.py

```diff
@@ -10,16 +10,34 @@
 
 
 class Mapping(Model):
-    def __init__(self, mapping):
+    """
+    Allows inputs to be reordered, duplicated or dropped.
+
+    Parameters
+    ----------
+    mapping : tuple
+        Integers representing indices of the inputs.
+
+    Raises
+    ------
+    TypeError
+        Raised when number of inputs is less that `max(mapping)`.
+
+    Examples
+    --------
+    >>> poly1 = Polynomial2D(1, c0_0=1, c1_0=2, c0_1=3)
+    >>> poly2 = Polynomial2D(1, c0_0=1, c1_0=2.4, c0_1=2.1)
+    >>> model = (Shift(1) & Shift(2)) | Mapping((0, 1, 0, 1)) | (poly1 & poly2)
+    >>> model(1, 2)
+    (17.0, 14.2)
+
+    """
+    def __init__(self, mapping, **kwargs):
         self._inputs = tuple('x' + str(idx)
                              for idx in range(max(mapping) + 1))
         self._outputs = tuple('x' + str(idx) for idx in range(len(mapping)))
         self._mapping = mapping
-        super(Mapping, self).__init__()
-
-    @property
-    def name(self):
-        return 'Mapping({0})'.format(self.mapping)
```

Comment:
(In other words, I agree that the default name property on these models wasn't very useful, but it would be good if they had a `__repr__` that indicates what they do instead, but that's a separate issue).